### PR TITLE
Remvoval of variadic functions and setjmp/longjmp in bibtex

### LINKED ISSUE #9: Replace setjmp/longjmp with catch_panic
The c code uses setjmp/longjmp in several places. They're not expressible in current rust. So we want to change it to using panic! and catch_panic.

### PR DESCRIPTION
**Pull request by mulimoen**
fixes #9

### CODE REVIEW

**Comment by cormacrelf** [comment]
Have you tested it with malformed bibtex?

**Comment by mulimoen** [comment]
Tried processing a malformed input, which gives the same stdout as the main branch.

It would be good to add a unit test to tectonic/tests/bibtex.rs with such a malformed unit and compare the (stdout) output, but I do not know how to get this from the backend.

**Comment by cormacrelf** [comment]
I don’t have a problem with setting up [insta](https://docs.rs/insta/0.11.0/insta/) tests on tectonic. Could even put pdf output in them, unsure if binary sections would trip up the harness too much but worth a shot.

**Comment by cormacrelf** [comment]
Idea in some more detail for PDFs: It will be w bit annoying to write the first run because of the file based api for tectonic vs the string or Serialize based one in insta, but would end up fine. Then a lib.rs/crates/datatest suite to test tests/*.tex. And a tweak in the harness to use tectonic upstream to snapshot new cases, so they start with a golden snapshot to compare to.

**Comment by mulimoen** [comment]
There is some integration testing which checks file equivalence in tectonic/tests

File: engine/src/bibtex.rs [hunk 1]
```diff
@@ -926,9 +869,9 @@ unsafe extern "C" fn bib_id_print() {
         puts_log(b"You\'re missing \x00" as *const u8 as *const i8);
     } else if scan_result as i32 == 2i32 {
         /*other_char_adjacent */
-        printf_log(
-            b"\"%c\" immediately follows \x00" as *const u8 as *const i8,
```
Comment (crlf0710) [inline engine/src/bibtex.rs:930]: The original format specifier uses %c. The current behavior changes since `buffer` is not a `String`.

Comment (mulimoen) [reply engine/src/bibtex.rs:930]: Changed all instances like this to print char

**Comment by crlf0710** [comment]
Thanks

**Pull request merged**